packages/schematics/src/command-line/nx.ts

```diff
@@ -1,49 +1,125 @@
 
-const processedArgs = yargsParser(process.argv, {
-  alias: {
-    app: ['a']
-  },
-  string: ['app']
-});
-const command = processedArgs._[2];
 const args = process.argv.slice(3);
 
-switch (command) {
-  case 'affected':
-    affected(args);
-    break;
-  case 'dep-graph':
-    generateGraph(yargsParser(args));
-    break;
-  case 'format':
-    format(args);
-    break;
-  case 'migrate': // TODO: delete this after 1.0
-    update(args);
-    break;
-  case 'update':
-    update(args);
-    break;
-  case 'lint':
-    lint();
-    break;
-  case 'postinstall':
-    patchNg();
-    update(['check']);
-    break;
-  case 'workspace-schematic':
-    workspaceSchematic(args);
-    break;
-  default:
-    throw new Error(`Unrecognized command '${command}'`);
-}
+yargs
+  .usage('Usage: $0 <command> [options>')
+  .command(
+    'affected',
+    'Compute affected applications',
+    yargs =>
+      yargs
+        .positional('action', {
+          choices: ['apps', 'build', 'e2e', 'dep-graph']
+        })
+        .command('<action> -- <SHA1 SHA2>', 'Targets within a range of SHAs')
+        .command(
+          '<action> -- files="<file,list>"',
+          'Targets a comma delimited list of files'
+        )
+        .command('<action> -- --uncommitted', 'Targets uncommited changes')
+        .command('<action> -- --untracked', 'Targets untracked changes')
+        .demandCommand()
+        .command(
+          'dep-graph',
+          'Generate a graph showing links between targets',
+          yargs =>
+            yargs
+              .describe('file', 'output file (e.g. --file=.vis/output.json)')
+              .command('-- <SHA1 SHA2>', 'Targets within a range of SHAs')
+              .command(
+                '-- files="<file,list>"',
+                'Targets a comma delimited list of files'
+              )
+              .command('-- --uncommitted', 'Targets uncommited changes')
+              .command('-- --untracked', 'Targets untracked changes')
+              .demandCommand()
+              .demand('file')
+              .demand('output')
+              .choices('output', [
+                OutputType.json,
+                OutputType.dot,
+                OutputType.html
+              ]),
+          _ => {
+            generateGraph(yargsParser(args));
+          }
+        )
+        .demandCommand()
+        .describe('args', 'Prints the names of the affected apps')
+        .describe('build', 'Builds the affected apps')
+        .describe('e2e', 'Tests the affected apps')
+        .describe(
+          'dep-graph',
+          'Generate a graph showing links between apps and libs'
+        ),
+    _ => {
+      affected(args);
+    }
+  )
+  .command(
+    'format <write|check>',
+    'Format workspace',
+    yargs =>
+      yargs
+        .describe('write', 'Overwrite non-formatted files')
+        .describe('check', 'Check if files need formatting'),
+    _ => {
+      format(args);
+    }
+  )
+  .command(
```

Comment:
I think we want to move away from using the `migrate` terminology. So maybe update the terms in the descriptions in here to be `update, updates` instead of `migrate, migrations`.